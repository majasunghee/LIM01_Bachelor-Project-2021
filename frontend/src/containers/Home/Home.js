/* eslint-disable react/jsx-wrap-multilines */
import React, { useState, useEffect } from 'react';
import axios from 'axios';
import {
  Button,
  Drawer,
  Hidden,
  List,
  ListItem,
  ListItemText,
  Divider,
} from '@material-ui/core';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import FavoriteIcon from '@material-ui/icons/Favorite';
import DoneIcon from '@material-ui/icons/Done';
import CollectionsBookmarkIcon from '@material-ui/icons/CollectionsBookmark';
import { Link, Redirect } from 'react-router-dom';
import { axiosInstanceDelete, axiosInstance } from '../../helpers/ApiFunctions';
import SearchBar from '../../components/SearchBar/SearchBar';
import SetCard from '../../components/SetCard';
import DeleteModal from '../../components/DeleteModal';
import useStyles from './styles';

const Home = () => {
  const classes = useStyles();

  const [ExerciseSetList, setExerciseSetList] = useState([]);
  const [savedList, setSavedList] = useState([]);
  const [completedList, setCompletedList] = useState([]);
  const [showSetType, setShowSetType] = useState(0);

  const [redirectPlay, setRedirectPlay] = useState(false);
  const [playId, setPlayId] = useState(null);

  const [open, setOpen] = useState(false);
  const [deleteId, setDeleteId] = useState(null);

  const [formDataEdit, setFormDataEdit] = useState(null);
  const [redirectEdit, setRedirectEdit] = useState(false);

  function getContent() {
    const requestOne = axiosInstance().get(`/usersets/`);
    const requestTwo = axiosInstance().get(`/saved/`);
    const requestThree = axiosInstance().get(`/usercompleted/`);
    axios
      .all([requestOne, requestTwo, requestThree])
      .then(
        axios.spread((...res) => {
          setExerciseSetList(res[0].data);
          setSavedList(res[1].data);
          setCompletedList(res[2].data);
        })
      )
      .catch((e) => {
        return e;
      });
  }

  useEffect(() => {
    getContent();
  }, []);

  function onDelete(id) {
    axiosInstanceDelete()
      .delete(`/deletesets/${id}`)
      .then(() => {
        setOpen(false);
        getContent();
      })
      .catch((e) => {
        return e;
      });
  }

  const renderSwitch = (param) => {
    switch (param) {
      case 0:
        return (
          <>
            {ExerciseSetList.map((set) => {
              return (
                <SetCard
                  type="mySet"
                  formData={set}
                  onClick1={() => {
                    setFormDataEdit(set);
                    setRedirectEdit(true);
                  }}
                  onClick2={() => {
                    setDeleteId(set.id);
                    setOpen(true);
                  }}
                />
              );
            })}
          </>
        );
      case 1:
        return (
          <>
            {savedList.map((saved) => {
              return (
                <SetCard
                  type="favorite"
                  formData={saved}
                  onClick1={() => {
                    setPlayId(saved.sets);
                    setRedirectPlay(true);
                  }}
                />
              );
            })}
          </>
        );
      case 2:
        return (
          <>
            {completedList.map((completed) => {
              return (
                <SetCard
                  type="completed"
                  formData={completed}
                  onClick1={() => {
                    setPlayId(completed.sets);
                    setRedirectPlay(true);
                  }}
                />
              );
            })}
          </>
        );
      default:
        return <> </>;
    }
  };

  return (
    <div className={classes.root}>
      <nav className={classes.drawer} aria-label="mailbox folders">
        {/* The implementation can be swapped with js to avoid SEO duplication of links. */}
        <Hidden xsDown implementation="css">
          <Drawer
            classes={{
              paper: classes.drawerPaper,
            }}
            variant="permanent"
            open
          >
            <div>
              <div className={classes.toolbar} />
              <List>
                <ListItem
                  button
                  component={Link}
                  to="/createexercise"
                  className={classes.drawerBtn}
                  style={{ textAlign: 'center' }}
                >
                  <ListItemText>Opprett oppgavesett</ListItemText>
                </ListItem>
              </List>
              <Divider />
              <List>
                {['Mine sett', 'Lagrede sett', 'Fullførte sett'].map(
                  (text, index) => (
                    <ListItem
                      button
                      selected={showSetType === index}
                      className={classes.drawerBtn}
                      key={text}
                      onClick={() => setShowSetType(index)}
                    >
                      <ListItemIcon>
                        {index === 0 && (
                          <CollectionsBookmarkIcon className={classes.svg} />
                        )}
                        {index === 1 && (
                          <FavoriteIcon className={classes.svg} />
                        )}
                        {index === 2 && <DoneIcon className={classes.svg} />}
                      </ListItemIcon>
                      <ListItemText primary={text} />
                    </ListItem>
                  )
                )}
              </List>
            </div>
          </Drawer>
        </Hidden>
      </nav>
      <div className={classes.content}>
        <SearchBar />
        <Hidden smUp implementation="css">
          <div className={classes.buttonList}>
            {['Mine sett', 'Lagrede sett', 'Fullførte sett'].map(
              (text, index) => (
                <Button
                  id={index}
                  key={text}
                  onClick={() => setShowSetType(index)}
                  variant="outlined"
                  size="small"
                  className={classes.drawerBtn}
                >
                  {text}
                </Button>
              )
            )}
          </div>
        </Hidden>
        <br />
        {renderSwitch(showSetType)}
      </div>
      {open && (
        <DeleteModal
          onDelete={() => onDelete(deleteId)}
          open={open}
          setOpen={setOpen}
        />
      )}
      {redirectEdit && (
        <Redirect
          to={{
            pathname: '/createexercise',
            state: { formSets: formDataEdit, editSet: true },
          }}
        />
      )}
      {redirectPlay && (
        <Redirect
          to={{
            pathname: '/sets',
            state: { id: playId },
          }}
        />
      )}
    </div>
  );
};

export default Home;
