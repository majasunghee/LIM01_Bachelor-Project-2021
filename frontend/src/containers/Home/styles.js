import { makeStyles } from '@material-ui/core/styles';

const drawerWidth = 240;

const useStyles = makeStyles((theme) => ({
  root: {
    height: '100%',
    display: 'flex',
    padding: theme.spacing(2),
    fontFamily: 'roboto, helvetica, arial, sansSerif',
  },
  drawer: {
    [theme.breakpoints.up('sm')]: {
      width: drawerWidth,
      flexShrink: 0,
    },
    display: 'block',
  },
  // necessary for content to be below app bar
  toolbar: theme.mixins.toolbar,
  drawerPaper: {
    width: drawerWidth,
    backgroundColor: 'inherit',
    alignItems: 'center',
    paddingTop: theme.spacing(3),
    display: 'inline',
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing(1),
    '& h3': {
      width: 'fit-content',
      margin: 'auto',
      padding: theme.spacing(1),
      textTransform: 'uppercase',
      color: 'mintcream',
    },
  },
  buttonList: {
    margin: theme.spacing(1),
    textAlign: 'center',
  },
}));
export default useStyles;
