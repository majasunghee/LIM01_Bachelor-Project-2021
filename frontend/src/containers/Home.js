import React from 'react';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  root: {
    alignItems: 'center',
    color: 'black',
    background: 'linear-gradient(90deg, #53A77A 1.46%, #80D197 100%)',
    height: '100%',
    flexDirection: 'column',
    padding: theme.spacing(3),
    fontFamily: 'roboto, helvetica, arial, sansSerif',
  },
  infoBox: {
    margin: 'auto',
    padding: theme.spacing(2),
    width: '50%',
  },
}));

const Home = () => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <div className={classes.infoBox} />
    </div>
  );
};

export default Home;
