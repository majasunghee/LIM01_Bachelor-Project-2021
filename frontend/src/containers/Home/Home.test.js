/* eslint-disable no-undef */
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Home from './Home';

describe('The Home component', () => {
  it('should contain four clickable buttons', () => {
    const { getByText } = render(
      <Router>
        <Home />
      </Router>
    );

    const title = getByText('Opprett oppgavesett');
    fireEvent.click(screen.getAllByText('Mine sett')[1]);
    fireEvent.click(screen.getAllByText('Lagrede sett')[1]);
    fireEvent.click(screen.getAllByText('Fullf√∏rte sett')[1]);

    expect(title.tagName.toLowerCase()).toEqual('span');
  });
});
