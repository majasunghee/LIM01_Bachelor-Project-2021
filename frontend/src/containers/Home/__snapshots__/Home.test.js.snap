// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Home component should render 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-searchBar-2"
  >
    <div
      class="makeStyles-searchBox-10"
    >
      <form>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
          >
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-search-11 MuiFormControl-marginDense MuiFormControl-fullWidth"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                  placeholder="Tast inn settets ID"
                  required=""
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-15 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-16"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-12 MuiButton-containedPrimary MuiButton-fullWidth"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                SPILL
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
  <div
    class="makeStyles-root-1"
  >
    <nav
      aria-label="mailbox folders"
      class="makeStyles-drawer-3"
    >
      <div
        class="PrivateHiddenCss-xsDown-21"
      >
        <div
          class="MuiDrawer-root MuiDrawer-docked"
        >
          <div
            class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-5 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
          >
            <div>
              <div
                class="makeStyles-toolbar-4"
              />
              <ul
                class="MuiList-root MuiList-padding"
              >
                <a
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root makeStyles-drawerBtn-8 MuiListItem-gutters MuiListItem-button"
                  href="/createexercise"
                  role="button"
                  style="text-align: center; border: 1px solid #367051;"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      Opprett oppgavesett
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </a>
              </ul>
              <hr
                class="MuiDivider-root"
              />
              <ul
                class="MuiList-root MuiList-padding"
              >
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root makeStyles-drawerBtn-8 MuiListItem-gutters MuiListItem-button Mui-selected"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root makeStyles-svg-9"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6z"
                      />
                      <path
                        d="M20 2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 10l-2.5-1.5L15 12V4h5v8z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      Mine sett
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root makeStyles-drawerBtn-8 MuiListItem-gutters MuiListItem-button"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root makeStyles-svg-9"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      Lagrede sett
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root makeStyles-drawerBtn-8 MuiListItem-gutters MuiListItem-button"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root makeStyles-svg-9"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      Fullførte sett
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="makeStyles-content-6"
    >
      <div
        class="PrivateHiddenCss-smUp-23"
      >
        <div
          class="makeStyles-buttonList-7"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-drawerBtn-8 MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
            id="0"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Mine sett
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-drawerBtn-8 MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
            id="1"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Lagrede sett
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-drawerBtn-8 MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
            id="2"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Fullførte sett
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <br />
    </div>
  </div>
</DocumentFragment>
`;
